src/System.Private.CoreLib/shared/System/Number.Parsing.cs

```diff
@@ -1740,6 +1771,29 @@ internal static unsafe bool TryParseSingle(ReadOnlySpan<char> value, NumberStyle
                 {
                     result = float.NaN;
                 }
+                else if (valueTrim.StartsWith(info.PositiveSign, StringComparison.OrdinalIgnoreCase))
+                {
+                    valueTrim = valueTrim.Slice(info.PositiveSign.Length);
+
+                    if (valueTrim.EqualsOrdinalIgnoreCase(info.PositiveInfinitySymbol))
```

Comment:
This is an interesting case, since if the user creates a custom culture where `PositiveInfinitySymbol` is set to something like `"+Infinity"`, we will actually now successfully parse `"++Infinity"`; I'm not sure it's a case we need/should worry about however.

Reply:
There are real cultures that do that. E.g.:

```csharp
            var f = new CultureInfo("wo").NumberFormat;
            Console.WriteLine(f.PositiveInfinitySymbol);
            Console.WriteLine(f.NegativeInfinitySymbol);
```

Result:

```
+Infini
-Infini
```

Would it make sense to check that the PositiveInfinitySymbol starts with PositiveSign already, and skip this part in the case?

Reply:
> Would it make sense to check that the PositiveInfinitySymbol starts with PositiveSign already, and skip this part in that case?

Yes, if we have existing cases where `PositiveInfinitySymbol` is prefixed with `PositiveSymbol` I think that makes sense.

Reply:
Fixed.